backend/municipal-services/ingestion-service/app/processor/factory/vendor_data_processor_factory.py

```
@@ -0,0 +1,49 @@
+        data_loader = add_validation_columns(data_loader)
+
+        if mdms_url:
+            mdms_client = MDMSClient(mdms_url)
+            try:
+                schema = mdms_client.fetch_vendor_schema(request_info)
+                validators.append(RequiredFieldValidator(schema.mdms[0].data.columns))
+                validators.append(PatternValidator(schema.mdms[0].data.columns))
+                # validators.append(IdentifierValidator())
+            except Exception as e:
+                print(f"Error: Could not set up MDMS validators - {e}")
+                raise Exception("Could not set up MDMS validators")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)